.github/workflows/build.yml

```diff
@@ -42,14 +42,19 @@ jobs:
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Run image
+        uses: abatilo/actions-poetry@v2.0.0
+        with:
+          poetry-version: "1.1.4"
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip tox
+          poetry install
+          poetry run pip install django==${{ matrix.django-version }}
```

Comment:
How about adding a cache so that project dependencies can be pulled from there instead of PyPI?

```
      - name: Configure poetry
        shell: bash
        run: python -m poetry config virtualenvs.in-project true

      - name: Set up cache
        uses: actions/cache@v2
        id: cache
        with:
          path: .venv
          key: venv-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('**/poetry.lock') }}
```

Reply:
Nice one! thanks